docs/configuration.md

```diff
@@ -131,14 +126,90 @@ ignore = ["E501"]
+    `pyproject.toml` files that lack a `[tool.ruff]` section.
+1. If a configuration file is passed directly via `--config`, those settings are used for _all_
+    analyzed. files, and any relative paths in that configuration file (like `exclude` globs or
+    `src` paths) are resolved relative to the _current_ working directory.
+1. If no `pyproject.toml` file is found in the filesystem hierarchy, Ruff will fall back to using
+    a default configuration. If a user-specific configuration file exists
+    at `${config_dir}/ruff/pyproject.toml`, that file will be used instead of the default
+    configuration, with `${config_dir}` being determined via the [`dirs`](https://docs.rs/dirs/4.0.0/dirs/fn.config_dir.html)
+    crate, and all relative paths being again resolved relative to the _current working directory_.
+1. Any `pyproject.toml`-supported settings that are provided on the command-line (e.g., via
+    `--select`) will override the settings in _every_ resolved configuration file.
+
+Unlike [ESLint](https://eslint.org/docs/latest/user-guide/configuring/configuration-files#cascading-and-hierarchy),
+Ruff does not merge settings across configuration files; instead, the "closest" configuration file
+is used, and any parent configuration files are ignored. In lieu of this implicit cascade, Ruff
+supports an [`extend`](settings.md#extend) field, which allows you to inherit the settings from another
+`pyproject.toml` file, like so:
+
+```toml
+[tool.ruff]
+# Extend the `pyproject.toml` file in the parent directory...
+extend = "../pyproject.toml"
+
+# ...but use a different line length.
+line-length = 100
+```
+
+All of the above rules apply equivalently to `ruff.toml` and `.ruff.toml` files. If Ruff detects
+multiple configuration files in the same directory, the `.ruff.toml` file will take precedence over
+the `ruff.toml` file, and the `ruff.toml` file will take precedence over the `pyproject.toml` file.
+
+## Python file discovery
+
+When passed a path on the command-line, Ruff will automatically discover all Python files in that
+path, taking into account the [`exclude`](settings.md#exclude) and [`extend-exclude`](settings.md#extend-exclude)
+settings in each directory's `pyproject.toml` file.
+
+By default, Ruff will also skip any files that are omitted via `.ignore`, `.gitignore`,
+`.git/info/exclude`, and global `gitignore` files (see: [`respect-gitignore`](settings.md#respect-gitignore)).
+
+Files that are passed to `ruff` directly are always linted, regardless of the above criteria.
+For example, `ruff check /path/to/excluded/file.py` will always lint `file.py`.
+
+## Jupyter Notebook discovery
+
+Ruff has built-in support for linting [Jupyter Notebooks](https://jupyter.org/).
+
+To opt in to linting Jupyter Notebook (`.ipynb`) files, add the `*.ipynb` pattern to your
+[`include`](settings.md#include) setting, like so:
+
+```toml
+[tool.ruff]
+include = ["*.py", "*.pyi", "**/pyproject.toml", "*.ipynb"]
+```
```

Comment:
Do we want to recommend `extend-include` instead and wherever this is mentioned?